drgn/helpers/linux/mm.py

```diff
@@ -1377,3 +1377,20 @@ def totalram_pages(prog: Program) -> int:
         return prog["_totalram_pages"].counter.value_()
     except KeyError:
         return prog["totalram_pages"].value_()
+
+
+@takes_program_or_default
+def in_direct_map(prog: Program, addr: IntegerLike) -> bool:
+    """
+    Return True if an address is within the kernel's direct memory mapping
+    :param addr: address to check
+    """
+    addr = operator.index(addr)
+    try:
+        start_addr = pfn_to_virt(prog["min_low_pfn"]).value_()
+        end_addr = (pfn_to_virt(prog["max_low_pfn"]) + prog["PAGE_SIZE"]).value_()
+        return start_addr <= addr < end_addr
+    except NotImplementedError:
```

Comment:
Don't bother catching this exception. The helper should fail hard if virtual address translation isn't supported like the other mm helpers.

Reply:
Would you like this helper's callers to fail on this as well, or should I push that exception handler up to each of the three callers?

Reply:
Let's keep it the way it currently is: `_identify_kernel_address()` should catch it, but the SLAB helpers should fail.